rust/src/nasl/builtin/network/socket.rs

```diff
@@ -534,81 +358,125 @@ impl NaslSockets {
+                            None,
+                            timeout,
+                            bufsz,
+                            get_retry(context),
+                        ) {
+                            // TODO: Set port transport
+                            tcp
+                        } else {
+                            continue;
+                        }
                     }
-                    _ => {
-                        return Err(FunctionErrorKind::WrongArgument(format!(
-                            "unsupported transport layer: {transport}{tls_version}"
-                        )))
+                    // Unsupported transport layer
+                    None | Some(OpenvasEncaps::Max) => {
+                        return Err(FunctionErrorKind::Diagnostic(
+                            format!("unsupported transport layer: {transport} (unknown)"),
+                            None,
+                        ))
                     }
-                },
-            }
+                    // TLS/SSL
+                    Some(tls_version) => match tls_version {
+                        OpenvasEncaps::Tls12 | OpenvasEncaps::Tls13 => {
+                            if let Ok(tls_conf) = Self::get_tls_conf(context) {
+                                if let Ok(tls) = create_tls_client(
+                                    vhost,
+                                    &tls_conf.cert_path,
+                                    &tls_conf.key_path,
+                                    &tls_conf.password,
+                                    &tls_conf.cafile_path,
+                                ) {
+                                    if let Ok(tcp) = TcpConnection::connect(
+                                        addr,
+                                        port,
+                                        Some(tls),
+                                        timeout,
+                                        bufsz,
+                                        get_retry(context),
+                                    ) {
+                                        tcp
+                                    } else {
+                                        continue;
+                                    }
+                                } else {
+                                    continue;
+                                }
+                            } else {
+                                continue;
+                            }
+                        }
+                        _ => {
+                            return Err(FunctionErrorKind::Diagnostic(
+                                format!("unsupported transport layer: {transport} {tls_version}"),
+                                None,
+                            ))
+                        }
+                    },
+                }
+            };
+            let fd = self.add(NaslSocket::Tcp(Box::new(tcp)));
+            fds.push(fd);
         }
```

Comment:
I find this block hard to read.

I suggest extracting into a sub-function and doing something like this to make the logic a little clearer:

```suggestion rust
        let sockets: Vec<Option<NaslSocket>> = vhosts
            .iter()
            .map(|vhost| {
                self.open_sock_tcp_vhost(context, addr, timeout, bufsz, port, vhost, transport)
            })
            .collect::<Result<_, _>>()?;

        Ok(NaslValue::Fork(
            sockets
                .into_iter()
                .filter_map(|socket| socket)
                .map(|socket| {
                    let fd = self.add(socket);
                    NaslValue::Number(fd as i64)
                })
                .collect(),
        ))
    }

    fn make_tls_client_connection(
        &self,
        context: &Context,
        vhost: &str,
    ) -> Option<ClientConnection> {
        Self::get_tls_conf(context).ok().and_then(|conf| {
            create_tls_client(
                vhost,
                &conf.cert_path,
                &conf.key_path,
                &conf.password,
                &conf.cafile_path,
            )
            .ok()
        })
    }

    fn open_sock_tcp_vhost(
        &self,
        context: &Context,
        addr: IpAddr,
        timeout: Duration,
        bufsz: Option<usize>,
        port: u16,
        vhost: &str,
        transport: i64,
    ) -> Result<Option<NaslSocket>, FunctionErrorKind> {
        if transport < 0 {
            // TODO: Get port transport and open connection depending on it
            todo!()
        }
        let tls = match OpenvasEncaps::from_i64(transport) {
            // Auto Detection
            Some(OpenvasEncaps::Auto) => {
                // Try SSL/TLS first
                self.make_tls_client_connection(context, vhost)
            }
            // IP
            Some(OpenvasEncaps::Ip) => None,
            // Unsupported transport layer
            None | Some(OpenvasEncaps::Max) => {
                return Err(FunctionErrorKind::Diagnostic(
                    format!("unsupported transport layer: {transport} (unknown)"),
                    None,
                ))
            }
            // TLS/SSL
            Some(tls_version) => match tls_version {
                OpenvasEncaps::Tls12 | OpenvasEncaps::Tls13 => {
                    self.make_tls_client_connection(context, vhost)
                }
                _ => {
                    return Err(FunctionErrorKind::Diagnostic(
                        format!("unsupported transport layer: {transport} {tls_version}"),
                        None,
                    ))
                }
            },
        };
        Ok(
            TcpConnection::connect(addr, port, tls, timeout, bufsz, get_retry(context))
                .map(|tcp| NaslSocket::Tcp(Box::new(tcp)))
                .ok(),
        )
    }
```